consensus/src/quorum_store/proof_coordinator.rs

```diff
@@ -74,6 +76,9 @@ impl IncrementalProofState {
                     .is_none()
                 {
                     self.aggregated_voting_power += voting_power as u128;
+                    if signer == self.info.author() {
+                        self.self_voted = true;
```

Comment:
this is a bit ugly, we can restrict the init to be from self vote only

Reply:
But the concern was that the remote sends us a vote before we do

Reply:
yes, but isn't that very unlikely given it needs to go through a network round trip compared to local channels?

Reply:
This is rare but not extremely unlikely, given that on both local and remote the batch has to be persisted. It was observed in Forge, where the network trip is short, and our other networks do have at least some peers with short round trips.

Reply:
hmm that's very surprising